Honeycomb Bravo.lua

```diff
@@ -116,396 +133,617 @@ function array_has_true(array)
-		doors_array[1] = dataref_table('anim/door/FR')
-		doors_array[2] = dataref_table('anim/door/ML')
-		doors_array[3] = dataref_table('anim/door/MR')
-		doors_array[4] = dataref_table('anim/door/BL')
-		doors_array[5] = dataref_table('anim/door/BR')
-		doors_array[6] = dataref_table('1-sim/anim/doors/cargoBack/anim')
-		doors_array[7] = dataref_table('1-sim/anim/doors/cargoSide/anim')
-		doors_array[8] = dataref_table('anim/doorC') -- 757 freighter cargo door
-	elseif PROFILE == "FF/767" then
-		doors_array[0] = dataref_table('1-sim/anim/doors/FL/anim')
-		doors_array[1] = dataref_table('1-sim/anim/doors/FR/anim')
-		doors_array[2] = dataref_table('1-sim/anim/doors/BL/anim')
-		doors_array[3] = dataref_table('1-sim/anim/doors/BR/anim')
-		doors_array[4] = dataref_table('1-sim/anim/doors/cargoFront/anim')
-		doors_array[5] = dataref_table('1-sim/anim/doors/cargoBack/anim')
-		doors_array[6] = dataref_table('1-sim/anim/doors/cargoSide/anim')
-	end
 
-	function handle_led_changes()
-		if bus_voltage[0] > 0 then
-			master_state = true
-
-			-- HDG & NAV
-			if bitwise.band(autopilot_state[0], 2) > 0 then
-				-- Heading Select Engage
-				set_led(LED.FCU_HDG, true)
-				set_led(LED.FCU_NAV, false)
-			elseif bitwise.band(autopilot_state[0], 512) > 0 or bitwise.band(autopilot_state[0], 524288) > 0 then
-				-- Nav Engaged or GPSS Engaged
-				set_led(LED.FCU_HDG, false)
-				set_led(LED.FCU_NAV, true)
-			elseif ONLY_USE_AUTOPILOT_STATE then
-				-- Aircraft known to use autopilot_state
-				set_led(LED.FCU_HDG, false)
-				set_led(LED.FCU_NAV, false)
+-- Landing gear LEDs
+local gear = dataref_table('sim/flightmodel2/gear/deploy_ratio')
+local retractable_gear = dataref_table('sim/aircraft/gear/acf_gear_retract')
+
+-- Annunciator panel - top row
+local master_warn = dataref_table('sim/cockpit2/annunciators/master_warning')
+local fire = dataref_table('sim/cockpit2/annunciators/engine_fires')
+local oil_low_p = dataref_table('sim/cockpit2/annunciators/oil_pressure_low')
+local fuel_low_p = dataref_table('sim/cockpit2/annunciators/fuel_pressure_low')
+local anti_ice = dataref_table('sim/cockpit2/annunciators/pitot_heat')
+local anti_ice_flip = false
+local starter = dataref_table('sim/cockpit2/engine/actuators/starter_hit')
+local apu = dataref_table('sim/cockpit2/electrical/APU_running')
+if PROFILE == "B738" then
+	fire = dataref_table('laminar/B738/annunciator/six_pack_fire')
+	fuel_low_p = dataref_table('laminar/B738/annunciator/six_pack_fuel')
+	anti_ice = dataref_table('laminar/B738/annunciator/six_pack_ice')
+elseif PROFILE == "FF/757" or PROFILE == "FF/767" then
+	master_warn = dataref_table('inst/loopwarning')
+	fire = dataref_table('sim/cockpit/warnings/annunciators/engine_fires')
+	anti_ice = dataref_table('sim/cockpit2/ice/ice_window_heat_on')
+	anti_ice_flip = true
+elseif PROFILE == "Toliss/32x" then
+	master_warn = dataref_table('AirbusFBW/MasterWarn')
+	apu_fire = dataref_table('AirbusFBW/OHPLightsATA26')
```

Comment:
I think we need to declare these as `local` otherwise they'll be global? I'm not super proficient at Lua :-/ Perhaps we should use an approach here like I did for `doors_array` where we put these various fire-related dataref tables into one array and then loop over that.